yogstation/code/modules/power/nuclear_engine.dm

```diff
@@ -0,0 +1,449 @@
+	STOP_PROCESSING(SSmachines,src)
+
+/obj/machinery/power/NuclearReactor/proc/CheckPipes() //In case your pipes blow up
+	outlet = null
+	outlet = locate(/obj/machinery/atmospherics/components/binary/pump) in get_step(src, NORTH)
+	if(outlet)
+		say("Success! outlet pump registered as [outlet]")
+		return TRUE
+	else
+		say("Error! No outlet pump could be found!")
+		return FALSE
+
+/obj/machinery/power/NuclearReactor/Destroy()
+	disconnect_from_network()
+	. = ..()
+
+/obj/machinery/power/NuclearReactor/proc/start()
+	if(!outlet)
+		say("ERROR! No outlet found!, please attach a standard atmospherics pump on the highlighted tile to the left!")
+	if(!powernet)
+		connect_to_network()
+	if(state & FUELHATCH_OPEN || state & WASTEHATCH_OPEN)
+		say("Warning! A hatch is still open! Please close it before trying this!")
+		return
+	say("Reaction started!")
+	Heat += 1 //kickstart it
+	START_PROCESSING(SSmachines,src)
+
+/obj/machinery/power/NuclearReactor/process()
+	if(ReactorInoperable)
+		stop()
+		return
+	UpdateIcon()
+	ProcessAtmos()
+	CheckState()
+	Time ++
+	react.ConsumeFuel()
+//	Heat = GetHeat()
+	//Efficiency = GetEfficiency() gonna play round with this temporarily
+	HeatRate = react.GetFuelAmounts()
+	Heat -= react.GetCoolAmounts()
+	if(HeatRate > 0)
+		Heat += HeatRate
+	if(Heat <= 0)
+		stop()
+		return
+
+/obj/machinery/power/NuclearReactor/proc/MeltdownAnnounce() //Announce the meltdown to give them a chance to RUN
+	stop()
+	ReactorInoperable = TRUE
+	if(ReactorInoperable)
+		for(var/mob/M in GLOB.player_list)
+			if(M.z == z)
+				M << null //Don't want to assault their eardrums
+				M << MeltDownSound
+				priority_announce("Nuclear reactor status: CRITICAL Meltdown imminent! Evacuate engineering section IMMEDIATELY", "Nuclear Reactor Monitoring Subsystem",'sound/ai/attention.ogg')
+				addtimer(CALLBACK(src, .proc/Meltdown), 410) //It's the final countdown
+
+/obj/machinery/power/NuclearReactor/proc/Meltdown() //qdel(station)
+	explosion(get_turf(src),20,40,40, 100)
+	var/datum/round_event_control/radiation_storm/RS = new()
+	RS.runEvent()
+	ReactorInoperable = TRUE
+	fuel = null
+	monitor = null
+	waste = null
+	cut_overlays()
+	icon_state = "reactor-broken"
+	for(var/turf/T in orange(src, 10))
+		new /obj/effect/nuclearwaste(T)
+
+
+/obj/machinery/power/NuclearReactor/proc/CheckState()
+	if(state & FUELHATCH_OPEN || state & WASTEHATCH_OPEN)
+		radiation_pulse(src, Heat, 40) //You fucked up kid
+	if(Heat >= MELTDOWN) //Oh FUCK
+		stop()
+		MeltdownAnnounce()
+		return
+	if(Heat >= CRITICAL_HEAT)
+		radiation_pulse(src, Heat*2, 60)
+		if(world.time >= SavedWarningTime + WarningSoundCooldown)
+			SavedWarningTime = world.time
+			for(var/mob/M in GLOB.player_list)
+				if(M.z == z)
+					M << WarningSound
+		else
+			return
+
+/obj/machinery/power/NuclearReactor/proc/ProcessAtmos()
+	for(var/datum/gas_mixture/S in outlet.airs)
+		if(S.temperature <= 20000) //A small nerf to avoid tritium production
```

Comment:
You sure this will work? Tritium production depends on the ratio between oxygen and plasma, not temperature